File: Hosts/Silverlight/Tests/tests/integration/test_execfile.rb
describe "exec file" do
  should 'define methods from execfile' do
    python "execfile('#{File.dirname(__FILE__)}/fixtures/mod1.py')"
    python('proof1()', :expression).should.equal "MOD1".to_clr_string
  end

  should 'define methods from a exec' do
    python "exec \"def proof2(): return \\\"MOD2\\\"\""
    python('proof2()', :expression).should.equal "MOD2".to_clr_string
  end
end

File: Hosts/Silverlight/Tests/tests/regressions/test_execfile.py
import unittest

class ExecFileRegression(unittest.TestCase):

    # FIXME: NameError: global name 'proof' is not defined
    def test_definemethodFrom_execfile(self):
        execfile('regressions/fixtures/mod1.py')
        #self.assertEqual(proof(), 'MOD1')
        #self.assertEqual(proof1(), 'MOD1')

    def test_definemethodFrom_exec(self):
        exec "def proof2(): return \"MOD2\"\ndef proof(): return \"MOD2\""
        self.assertEqual(proof(), 'MOD2')
        self.assertEqual(proof2(), 'MOD2')

